File: 199. Binary Tree Right Side View/Python/Solution.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution(object):
    def rightSideView(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        if not root:
            return []
        queue = collections.deque()
        result = []
        queue.append(root)
        while queue:
            result.append(queue[0].val)
            size = len(queue)
            for i in range(size):
                node = queue.popleft()
                if node.right:
                    queue.append(node.right)
                if node.left:
                    queue.append(node.left)
        return result

File: 199. Binary Tree Right Side View/C++/Solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> rightSideView(TreeNode *root) {
        if (root == nullptr) {
            return vector<int>();
        }
        queue<TreeNode *> que;
        vector<int> result;
        que.push(root);
        while (!que.empty()) {
            result.push_back(que.front()->val);
            int size = que.size();
            for (int i = 0; i < size; i++) {
                TreeNode *node = que.front();
                que.pop();
                if (node->right != nullptr) {
                    que.push(node->right);
                }
                if (node->left != nullptr) {
                    que.push(node->left);
                }
            }
        }
        return result;
    }
};